tests/src/3d/testqgs4978rendering.cpp

```diff
@@ -0,0 +1,212 @@
+/***************************************************************************
+  testqgs4978rendering.cpp
+  --------------------------------------
+  Date                 : May 2022
+  Copyright            : (C) 2022 by Benoit De Mezzo
+  Email                : benoit dot de dot mezzo at oslandia dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgstest.h"
+#include "qgsmultirenderchecker.h"
+
+#include "qgslinestring.h"
+#include "qgsproject.h"
+#include "qgsvectorlayer.h"
+#include "qgsapplication.h"
+#include "qgs3d.h"
+
+#include "qgs3dmapscene.h"
+#include "qgs3dmapsettings.h"
+#include "qgs3dutils.h"
+#include "qgscameracontroller.h"
+#include "qgsflatterraingenerator.h"
+#include "qgsline3dsymbol.h"
+#include "qgsoffscreen3dengine.h"
+#include "qgsrulebased3drenderer.h"
+#include "qgsterrainentity_p.h"
+#include "qgsvectorlayer3drenderer.h"
+#include "qgs25drenderer.h"
+#include "qgssimplelinematerialsettings.h"
+#include "qgslinesymbol.h"
+#include "qgssinglesymbolrenderer.h"
+
+#include <QFileInfo>
+#include <QDir>
+#include <QDesktopWidget>
+
+class TestQgs4978Rendering : public QObject
+{
+    Q_OBJECT
+
+  private slots:
+    void initTestCase();// will be called before the first testfunction is executed.
+    void cleanupTestCase();// will be called after the last testfunction was executed.
+    void test3dLineRendering();
+    void test2dLineRendering();
+
+  private:
+    bool render3dCheck( const QString &testName, QImage &image, int mismatchCount = 0 );
+    bool render2dCheck( const QString &testName, QgsVectorLayer *layer, QgsRectangle extent, QgsCoordinateReferenceSystem *crs = nullptr );
+
+    QString mReport;
+
+    std::unique_ptr<QgsProject> mProject;
+
+};
+
+//runs before all tests
+void TestQgs4978Rendering::initTestCase()
```

Comment:
Can you add this test to the existing testqgs3drendering.cpp file instead? We should try to avoid adding new cpp tests as much as reasonable, as each one adds considerably to the compile/linking times.

Reply:
I have put these tests apart because some of them are only 2d tests. It does not seem logical to me to put them with the 3d tests. Do you still want to put them with the 3D tests?

Reply:
> Do you still want to put them with the 3D tests?

Or split them into other existing 2d/3d tests. I'm just against introducing new .cpp tests solely for one specific issue.

Reply:
ok, I will split them